exporters/otlp/otlplog/otlploggrpc/config.go

```diff
@@ -4,28 +4,58 @@
 package otlploggrpc // import "go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploggrpc"
 
 import (
+	"fmt"
 	"time"
 
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials"
 
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploggrpc/internal/otlpconf"
 	"go.opentelemetry.io/otel/exporters/otlp/otlplog/otlploggrpc/internal/retry"
 )
 
+type wrappedOption struct {
+	otlpconf.Option
+}
```

Comment:
There is a lot of boilerplate introduced by having configuration live in a separate package. Including this and all the duplicate options patterns. As @pellared pointed out, merging this into a single configuration shared between the HTTP exporter and this is undesirable.

Instead, can we just implement the configuration within `otlploggrpc` and be more direct?

Reply:
> Instead, can we just implement the configuration within otlploggrpc and be more direct?

We can, but I have a concern about copying similar configuration too much like what we have in the https://github.com/open-telemetry/opentelemetry-go/blob/1cc062a736dce3639358d7c87222d83d3bcaffc5/exporters/otlp/otlplog/otlploggrpc/internal/otlpconf/setting.go file. Say, if we have a fix for TLS certificate implementation, we have to manually make two changes, one for `otlploghttp` and the other for `otlploggrpc`, as they do not use a shared template file. It also applies the tests that are not shared.

> Keep in mind that I am not sure if this would be worth the effort given these configs are not 1:1.

@pellared The way I implement can only put the implementation that both `otlploghttp` and `otlploggrpc` are needed, and these packages can have their own unique extension on config, like this https://github.com/open-telemetry/opentelemetry-go/blob/1cc062a736dce3639358d7c87222d83d3bcaffc5/exporters/otlp/otlplog/otlploggrpc/config.go#L220-L225

Reply:
Can we extract TLS certificate support so it is signal/transport agnostic and just share that?

Reply:
I found it tedious to abstract only TLS certificate methods without abstracting the entire package. Maybe I can do this in the following PRs after finishing the implementation of `otlogloggrpc`, but not in this PR.